storage/replicate_queue.go

```diff
@@ -162,17 +165,31 @@ func (rq *replicateQueue) process(
 		log.Event(ctx, "removing a replica")
 		// We require the lease in order to process replicas, so
 		// repl.store.StoreID() corresponds to the lease-holder's store ID.
-		removeReplica, err := rq.allocator.RemoveTarget(desc.Replicas, repl.store.StoreID())
+		removeReplica, err := rq.allocator.RemoveTarget(desc.Replicas, 0)
 		if err != nil {
 			return err
 		}
-		log.VEventf(1, ctx, "removing replica %+v due to over-replication", removeReplica)
-		if err = repl.ChangeReplicas(ctx, roachpb.REMOVE_REPLICA, removeReplica, desc); err != nil {
-			return err
-		}
-		// Do not requeue if we removed ourselves.
 		if removeReplica.StoreID == repl.store.StoreID() {
-			return nil
+			var targetID roachpb.StoreID
+			for _, replica := range desc.Replicas {
```

Comment:
Yeah, this was too naive. Finding the lease loaded store from the existing replicas (excluding the lease holder) is straightforward, though I'm unable to upload the change right now.